src/internal/_getTypedFunction.js

```diff
@@ -0,0 +1,6 @@
+var _FN_TYPE_KEY = require('./_FN_TYPE_KEY');
+
+module.exports = function _getTypedFunction(type, f) {
+    f[_FN_TYPE_KEY] = type;
+    return f;
+};
```

Comment:
Makes sense, what should I call it? _createTypedFunction or _makeTypedFunction or..?